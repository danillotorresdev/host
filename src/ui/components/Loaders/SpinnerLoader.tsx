import { styled } from "@linaria/react";

interface SpinnerProps {
  size?: number;
  color?: string;
}

const SpinnerWrapper = styled.div<SpinnerProps>`
  display: inline-block;
  width: ${(props) => props.size ?? 50}px;
  height: ${(props) => props.size ?? 50}px;
  border: 3px solid ${(props) => props.color ?? "#ccc"};
  border-radius: 50%;
  border-top-color: transparent;
  animation: spin 1s ease-in-out infinite;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const SpinnerLoader = ({ size, color }: SpinnerProps) => <SpinnerWrapper size={size} color={color} data-testid="spinner-loader" />;
