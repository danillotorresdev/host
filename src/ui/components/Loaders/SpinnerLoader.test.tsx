import { SpinnerLoader } from "./SpinnerLoader";
import { render } from "@/utils/settings/tests/utilities";
import { BrowserRouter } from "react-router-dom";
import { expect, it, describe } from "vitest";


describe("SpinnerLoader", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <SpinnerLoader />,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});