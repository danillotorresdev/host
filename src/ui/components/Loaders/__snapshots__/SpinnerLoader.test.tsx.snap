// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpinnerLoader > should render the user's name 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 s1h0i5n1"
    data-testid="spinner-loader"
    style="--s1h0i5n1-0: 50px; --s1h0i5n1-1: #ccc;"
  />
</DocumentFragment>
`;
